### PR TITLE
wifi : Bugfix for RRM, BTM request handling

### PR DESCRIPTION
**Pull request by sr1dh48r**
[SHEL-877]: DUT did not respond to AP beacon request frame 
Enable misisng RRM capability in Assoc request.
Update clock to fix BSS expiration issue.

[SHEL-859]: No BTM response for request from AP
Include PREV_BSSID flag for re-assoc
Use frequencies in params during scan
Handle deauth during BSS transition

Signed-off-by: Sridhar Nuvusetty <sridhar.nuvusetty@nordicsemi.no>

### CODE REVIEW

File: samples/wifi/shell/src/wifi_main.c [hunk 1]
```diff
@@ -15,4 +16,10 @@ void main(void)
 			       NRF_CLOCK_HFCLK_DIV_1);
 #endif
 	printk("Starting %s with CPU frequency: %d MHz\n", CONFIG_BOARD, SystemCoreClock/MHZ(1));
+
+	/* set clock to be above BSS EXPIRATION value */
+	struct timespec tspec;
+	tspec.tv_sec = 50000;
+	tspec.tv_nsec = 10000;
+	clock_settime(CLOCK_REALTIME, &tspec);
```
Comment (krish2718) [inline samples/wifi/shell/src/wifi_main.c:24]: This looks like a hack, any specific reason? I am debugging a similar bug related to BSS flush from Matter, is it the same?

Comment (krish2718) [reply samples/wifi/shell/src/wifi_main.c:24]: I have root cause and raised https://github.com/nrfconnect/sdk-hostap/pull/53 with proper fix (added your SOB too).

**Review by tejlmand — APPROVED** [review]
lgtm

**Review by sachinthegreen — APPROVED** [review]

**Review by krish2718 — CHANGES_REQUESTED** [review]
Based on above comments, we need to rebase this to include the BSS age fix.

**Review by krish2718 — APPROVED** [review]

**Pull request merged**